### ISSUE TITLE
AST/asm import via Standard JSON

### ISSUE THREAD

**Issue by cameel** [ethereum/solidity#13690]
Related to #13157.

Currently both the `--import-ast` option and the new assembly import option (#11787) only work on the CLI. We need Standard JSON equivalents.

The `--import-ast` option in particular has been in the compiler since 0.6.2 but while talking with tool developers at Devcon we discovered that they're mostly unaware of it even though it would have been helpful to them. One of the reasons is that it's not in Standard JSON.

**Comment by NunoFilipeSantos** [comment]
Low effort/High impact are some of the top issues we should tackle. 😉

**Comment by ekpyron** [comment]
Any particular open design questions here (seeing the "needs design" label)? (since @aarlt may be doing this after the assembly import is done)
We already had a quick talk about this in the past after which for the AST input, I'd basically make the standard json AST output and the input as symmetric as possible, which should be sufficient to clarify the input format.

**Comment by cameel** [comment]
Yeah, it needs a decision about the input format. It should be symmetric of course but we also need to place it somewhere in the Stardard JSON input structure. Basically, choose a key for it.

When we discussed it on the channel last time, my opinion was that we should introduce a new key that you can use instead of `sources`. E.g. `import` and `inputs`, with subkeys for various things we can import (matching output names). It should contain non-stringified JSON.

AST import should also only be allowed when `language` is set to `Solidity` (not `Yul`) since we don't support Yul AST import. With assembly - not sure. Maybe we should not have `language` in that case at all?

**Comment by ekpyron** [comment]
After our last chat, I'd actually have made ``SolidityAST`` the ``language`` :-). And then literally supported the exact ``sources`` field as in ``test/cmdlineTests/standard_only_ast_requested/output.json`` as ``sources`` field in the input. But yeah, also haven't thought through that yet.

**Comment by cameel** [comment]
This depends on how granular `language` is supposed to be. You could consider the AST JSON a separate language I guess but to me it also makes perfect sense to still treat it as `Solidity` input, just encoded differently. It goes through the same compiler stack with almost all the same options, has same outputs.

**Comment by cameel** [comment]
That's for the AST import. For the asm input I think a new language (`EVMASM`?) does make sense.

**Comment by cameel** [comment]
Looks like we already had an issue for AST import in Standard JSON, though the way it's titled at first it seems like it's about something else: #2166

**Comment by ekpyron** [comment]
> After our last chat, I'd actually have made `SolidityAST` the `language` :-). And then literally supported the exact `sources` field as in `test/cmdlineTests/standard_only_ast_requested/output.json` as `sources` field in the input. But yeah, also haven't thought through that yet.

As for the format: let's do it like this comment suggests, i.e. ``SolidityAST`` as language, and then a ``sources`` field as in the AST export output.

**Comment by aarlt** [comment]
I think it would be good to be as specific as possible. So I think that `SolidityAST` as the language would make sense. Also for the evm asm json import, I would also think here to be as much specific as possible cannot hurt, e.g. `EvmAssemblyJSON` as language for the evm import. However, I also think that having a symmetric format for `input` and `output` is the way to go.
